hubclient/client.go

```diff
@@ -576,3 +572,8 @@ func (c *Client) GetAuthTokenExpiryTime() int64 {
 	}
 	return c.authTokenExpiryInUnixSec
 }
+
+// Sets the User-Agent header value to be used in all http/https requests made by the client
+func (c *Client) SetUserAgent(agent string) {
```

Comment:
To make this thread safe, instead of this Set func we could have a new `hubclient.Client` initialization function that accepts and sets the User-Agent value once during client initialization.

Reply:
I'd like to leave this for now and revisit after we discuss priority.  Generally I expect User-Agent is set once by consumers and not touched, but having it be safe would be preferable. 

I exposed this specifically because it was the one usage I found in external code of the `SetHeader` method.

Reply:
> Generally I expect User-Agent is set once by consumers and not touched

Yeah..this is another good reason why we should make it part of  the initialization (constructor) function.

But I will leave it to you to prioritize this.